### 課題１ Assignment 1

#### 帰無仮説・対立仮説を立てる
- 帰無仮説: 今回行なった調査の満足度のスコアの母平均は50である．(今回行なった調査の満足度のスコアは過去の調査の満足度スコアと差がない)
- 対立仮説: 今回行なった調査の満足度のスコアの母平均は50でない．(今回行なった調査の満足度のスコアは過去の調査の満足度スコアと差がある)
- 両側検定
- 母分散は既知

Set up the null/alternative hypothesis 
- H0: The population mean of the satisfaction scores of the survey conducted this time is 50. (The satisfaction score of the survey conducted this time is not different from the satisfaction score of past surveys)
- H1: The population mean of the satisfaction scores of the survey conducted this time is not 50. (The satisfaction scores of the surveys conducted this time are different from the satisfaction scores of past surveys)
- two-sided test
- Z-test because the population variance is known

In [1]:
import numpy as np
import pandas as pd
import scipy.stats as ss

#### 母集団分布の平均，標準偏差
Population mean and standard deviation

In [2]:
mean =  50 # 母平均 population mean
std = 15 # 母標準偏差 population standard deviation

#### 標本の平均，サイズ
Sample mean and size

In [3]:
# Pandas
df = pd.read_csv("data/survey.csv",header=None) # 標本 sample
sample_mean = df[0].mean() #標本平均 sample mean
print(sample_mean)
n = df.shape[0] # 標本サイズ sample size
print(n)

# NumPy
#data = np.loadtxt("data/survey.csv")
#sample_mean = np.mean(data) #標本平均 sample mean
#print(sample_mean)
#n = len(data) # 標本サイズ sample size
#print(n)

49.544
1000


#### 棄却域
critical region

In [4]:
alpha = 0.05 # 有意水準　significant level
# 棄却域 critical region
z_lower = ss.norm.ppf(alpha/2) # 下側2.5%点 lower 2.5% point
z_upper = ss.norm.ppf(1-alpha/2) # 上側2.5%点 upper 2.5% point
print(z_lower, z_upper)

-1.9599639845400545 1.959963984540054


#### 検定統計量zを求める
Calculate the static z 

In [5]:
z = (sample_mean - mean)/(std / np.sqrt(n))
print(z)

-0.9613324086911939


**検定統計量(test statistic): -0.961**

#### 検定統計量zが棄却域に含まれているか調べる
Check whether the statistic z is in the critical region

In [6]:
if z_lower > z or z_upper < z: # 検定統計量zが棄却域に含まれていたら帰無仮説を棄却
    print("Reject H0 (帰無仮説を棄却)")
else:
    print("Retail H0 (帰無仮説を棄却できない)")

Retail H0 (帰無仮説を棄却できない)


#### p値を求める
Calculate p-value

In [8]:
p_val = ss.norm.cdf(-np.abs(z))*2 #p値 (両側検定) p-value (two-sided test)
print(p_val)

# p値が有意水準よりも小さければ帰無仮説を棄却　reject the null hypothesis if the p-value is less than the significance level
if p_val < alpha: 
    print("Reject H0 (帰無仮説を棄却)")
else:
    print("Retail H0 (帰無仮説を棄却できない)")

0.3363850576123141
Retail H0 (帰無仮説を棄却できない)


**p値: 0.336**

#### 結論 Results
**p値が有意水準0.05よりも大きいので帰無仮説を棄却されない．  
つまり，今回行なった調査の満足度のスコアは過去の調査の満足度スコアと差があるとはいえない**

**p-value > 0.05, 
so, it cannot to be said that the satisfaction scores of the surveys conducted this time are different from the satisfaction scores of past surveys**